src/Components/MaterialBiSelect.tsx

```diff
@@ -0,0 +1,63 @@
+import { Grid } from '@material-ui/core';
+import { AnimatePresence, motion } from 'framer-motion';
+import React from 'react';
+import { UserObject, ProblemObject, SettingsComponentType, StudentWorkbookInterface } from '../Courses/CourseInterfaces';
+import MultiSelectCardList from './MultiSelectCardList';
+
+interface MaterialBiSelectProps {
+    problems: ProblemObject[];
+    users: UserObject[];
+    selected: {
+        problem?: ProblemObject,
+        user?: UserObject,
+        workbook?: StudentWorkbookInterface,
+    };
+    setSelected: React.Dispatch<React.SetStateAction<{
+        problem?: ProblemObject,
+        user?: UserObject,
+        workbook?: StudentWorkbookInterface,
+    }>>
+}
+
+export const MaterialBiSelect: React.FC<MaterialBiSelectProps> = ({problems, users, selected, setSelected}) => {
+    const onItemClick = (type: SettingsComponentType) => {
+        if (type instanceof ProblemObject) {
+            setSelected({...selected, problem: type});
```

Comment:
Use a function when re-applying state
```suggestion
            setSelected(selected => ({...selected, problem: type}));
```

Reply:
is this an 'always' thing? I haven't seen any difference either way...

Reply:
done